auto_nag/scripts/topcrash_add_keyword.py

```diff
@@ -0,0 +1,165 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import re
+
+from libmozdata.bugzilla import Bugzilla
+from libmozdata.connection import Connection
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.topcrash import Topcrash
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+
+
+class TopcrashAddKeyword(BzCleaner):
+    def description(self):
+        return "Bugs with missing topcrash keywords"
+
+    def columns(self):
+        return ["id", "summary", "severity", "added_keyword"]
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        if bugid in data:
+            return
+
+        has_topcrash_keyword = "topcrash" in bug["keywords"]
+        has_startup_keyword = "topcrash-startup" in bug["keywords"]
+        if has_topcrash_keyword and has_startup_keyword:
+            return
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+        has_topcrash_signature = any(
+            signature in self.topcrashes for signature in signatures
+        )
+        assert has_topcrash_signature
+
+        has_startup_signature = any(
+            self.topcrashes.get(signature) for signature in signatures
+        )
+
+        should_add_keyword = (
+            has_topcrash_signature
+            and not (has_topcrash_keyword or has_startup_signature)
+        ) or (has_startup_signature and not has_startup_keyword)
+
+        if not should_add_keyword:
+            return
+
+        keyword_to_add = "topcrash-startup" if has_startup_signature else "topcrash"
+
+        autofix = {
+            "keywords": {
+                "add": keyword_to_add,
+            },
+            "comment": {
+                "body": "The bug is linked to a startup topcrash signature."
+                if has_startup_signature
+                else "The bug is linked to a topcrash signature.",
+            },
+        }
+
+        ni_person = utils.get_mail_to_ni(bug)
+        if ni_person and bug["severity"] not in HIGH_SEVERITY:
```

Comment:
Let's make enabling the needinfo optional (so if we find it to be noisy we can disable it without disabling the adding of keywords).

Reply:
I guess it doesn't have a lot of sense to make it optional as we have the other script asking to bump severity anyway...

Reply:
Yes, and if we need to disable it, it will be a simple change anyway.